src/Appwrite/Platform/Workers/Deletes.php

```diff
@@ -856,8 +858,8 @@ private function deleteFunction(Database $dbForPlatform, callable $getProjectDB,
          */
         Console::info("Deleting rules for function " . $functionId);
         $this->deleteByGroup('rules', [
-            Query::equal('resourceType', ['function']),
-            Query::equal('resourceInternalId', [$functionInternalId]),
+            Query::equal('type', ['deployment']),
+            Query::equal('automation', ['function=' . $functionId]),
```

Comment:
this part is a bit confusing when compared to https://github.com/appwrite/appwrite/blob/88ae06af57e14e3012e3873c4f7abc859f4aab02/src/Appwrite/Platform/Workers/Deletes.php#L796-L808

whats the difference for function vs sites?

Reply:
Rule can only 3 very simple things:
- Proxy to API
- Redirect
- Proxy to deployment (function or site, it doesnt care)

Due to this, link is always domain -> dpeloyment. Since creation, it ONLY points to one singular deployment.

To support auto-updating (for example when there is new active deployment, or new commit), we added automations:
- site=X updates deploymentId when there is new active site deployment
- function=X ^ same, but for functions
- branch=X updates when there is new deployment on that VCS branch

All of this is done by builds worker after successful build